infra/stream/test/TestFormatter.cpp

```diff
@@ -0,0 +1,241 @@
+#include "infra/stream/Formatter.hpp"
+#include "gtest/gtest.h"
+
+class TestFormatObject
+{
+public:
+    int v{ 9 };
+    bool b{ true };
+};
+template<>
+void infra::Formatter<TestFormatObject &>::Format(infra::StreamWriter& writer, infra::FormatSpec& spec)
+{
+    Formatter<decltype(value.v)>(value.v).Format(writer, spec);
+}
+
+class FormatTest : public testing::Test
+{
+public:
+    FormatTest()
+        : writer(streamWriter.Writer())
+    {}
+protected:
+    void AssertFormatter(const char* expected) const
+    {
+        EXPECT_EQ(expected, streamWriter.Storage());
+    }
+    infra::StringOutputStream::WithStorage<60> streamWriter;
+    infra::StreamWriter& writer;
+};
+TEST_F(FormatTest, emptyString)
+{
+    Format(writer, "");
+    AssertFormatter("");
+}
+TEST_F(FormatTest, simpleString)
+{
+    Format(writer, "a");
+    AssertFormatter("a");
+}
+TEST_F(FormatTest, missingArgument)
+{
+    Format(writer, "{}");
+    AssertFormatter("}");
+}
+TEST_F(FormatTest, outOfIndex)
+{
+    Format(writer, "{1}");
+    AssertFormatter("}");
+}
+TEST_F(FormatTest, stringWidth)
+{
+    Format(writer, "{:5}", "b");
+    AssertFormatter("b    ");
+}
+TEST_F(FormatTest, boolean)
+{
+    Format(writer, "{}", true);
+    AssertFormatter("true");
+}
+TEST_F(FormatTest, booleanCenter)
+{
+    Format(writer, "{:^14}", false);
+    AssertFormatter("    false     ");
+}
+
+TEST_F(FormatTest, integer0)
+{
+    Format(writer, "{0}", 0);
+    AssertFormatter("0");
+}
+TEST_F(FormatTest, uint8_t)
```

Comment:
uint8_t is a tricky test name. GTest converts it to a function that has a prefix, but this could potentially break. Use test naming convention to solve. Or better yet, use the data driven approach as suggested above.